server/jetstream_api.go

```diff
@@ -617,7 +616,8 @@ const JSApiConsumerLeaderStepDownResponseType = "io.nats.jetstream.api.v1.consum
 
 // JSApiLeaderStepdownRequest allows placement control over the meta leader placement.
 type JSApiLeaderStepdownRequest struct {
-	Placement *Placement `json:"placement,omitempty"`
+	PreferredServer string     `json:"preferred_server,omitempty"`
```

Comment:
We can't modify the original `Placement` struct without changing the API for stream configs etc, but I've added a new `PlacementForStepdown` struct that embeds it, so that all keys come under the JSON `placement` key.

Reply:
Have put it into `Placement` for now and guarded against it being set in stream/consumer configs.